design-toolkit/components/src/components/options/index.tsx

```diff
@@ -12,86 +12,185 @@
-  OptionsItem,
-  type OptionsItemProps,
-} from '../options-item';
+  ListBoxItem as AriaListBoxItem,
+  Text as AriaText,
+} from 'react-aria-components';
+import { Icon } from '../icon';
 
-export interface OptionsProps<T extends IOptionsItem>
-  extends AriaListBoxProps<T> {
-  className?: string;
-  description?: string;
-  errorMessage?: string;
-  label?: string;
-  placeholder?: string;
-  size?: OptionsItemProps<T>['size'];
-  type?: OptionsItemProps<T>['type'];
-}
+import type {
+  IOptionsItem,
+  OptionsItemIconProps,
+  OptionsItemProps,
+  OptionsItemTextProps,
+  OptionsProps,
+  OptionsSectionProps,
+} from './types';
+
+import { OptionsStyles } from './styles';
+
+const { list, section, sectionHeader, item, itemIcon, itemContent } =
+  OptionsStyles();
 
 export const OptionsContext = createContext<
-  Pick<OptionsProps<IOptionsItem>, 'size' | 'type'>
+  ContextValue<OptionsProps<IOptionsItem>, HTMLDivElement>
 >({
   size: 'large',
-  type: 'default',
+  color: 'default',
 });
 
 export function Options<T extends IOptionsItem>({
-  children,
-  className,
-  description,
-  errorMessage,
-  label,
-  placeholder,
-  size,
-  type,
+  ref,
   ...props
 }: OptionsProps<T>) {
+  [props, ref] = useContextProps(props, ref ?? null, OptionsContext);
+  const {
+    children,
+    className,
+    description,
+    errorMessage,
+    label,
+    placeholder,
+    size,
+    color,
+    ...rest
+  } = props;
   return (
-    <OptionsContext.Provider value={{ size, type }}>
-      <AriaListBox<T>
-        className='max-h-[200px] overflow-y-auto overflow-x-clip rounded-medium bg-surface-overlay shadow-elevation-overlay outline outline-static-light'
-        {...props}
-      >
+    <OptionsContext.Provider value={{ size, color }}>
+      <AriaListBox<T> ref={ref} className={list()} {...rest}>
         {children}
       </AriaListBox>
     </OptionsContext.Provider>
   );
 }
 Options.displayName = 'Options';
 
-Options.Item = OptionsItem;
-
-interface OptionsSectionProps<T extends IOptionsItem>
-  extends AriaListBoxSectionProps<T> {
-  header?: string;
-}
-
 export function OptionsSection<T extends IOptionsItem>({
   children,
   header,
   items,
 }: OptionsSectionProps<T>) {
   return (
-    <AriaListBoxSection
-      id={header}
-      className='mt-s border-default-dark border-t first:border-none'
-    >
-      <AriaHeader className='m-xs my-s text-default-dark text-header-xs'>
-        {header}
-      </AriaHeader>
+    <AriaListBoxSection id={header} className={section()}>
+      <AriaHeader className={sectionHeader()}>{header}</AriaHeader>
       <AriaListBoxCollection items={items}>{children}</AriaListBoxCollection>
     </AriaListBoxSection>
   );
 }
 OptionsSection.displayName = 'Options.Section';
+
+export function OptionsItem({
+  children,
+  className,
+  size: sizeProp,
+  color: colorProp,
+  ...props
+}: OptionsItemProps) {
+  const context = useContext(OptionsContext) ?? {};
+
+  const { size, color } = useMemo(
```

Comment:
Instead of mergingProps, you can use `isSlottedContext` to narrow the type of the context and fallback to the context value if it's not provided by the local prop. Like this: https://github.com/gohypergiant/standard-toolkit/blob/c16b4797d7ca4d3e081d72ae03a9fec2e580659c/design-toolkit/components/src/components/accordion/index.tsx#L74